crates/xflags-macros/src/emit.rs

```diff
@@ -380,103 +411,86 @@ fn emit_default_transitions(buf: &mut String, cmd: &ast::Cmd) {
+        w!(help_buf, "\nArguments:\n");
+        // let offset =
+        //     cmd.args_with_default().into_iter().map(|a| a.val.name.len() + 4).max().unwrap_or(2);
+        for arg in cmd.args_with_default() {
+            let (l, r) = arg.arity.brackets();
+            let pre_doc = format!("{l}{}{r}", arg.val.name);
+            // w_arg(&mut help_buf, offset, &pre_doc);
+            w!(help_buf, "  {:<20} {}\n", pre_doc, arg.doc.as_deref().unwrap_or(""));
         }
     }
-
-    let flags = cmd.flags_with_default();
-    if !flags.is_empty() {
-        if !empty_help {
-            blank_line(buf);
-        }
-        w!(buf, "{indent}OPTIONS:\n");
-
-        let mut blank = "";
-        for flag in &flags {
-            w!(buf, "{blank}",);
-            blank = "\n";
-
+    if !cmd.flags_with_default().is_empty() {
+        w!(help_buf, "\nOptions:\n");
+        // let offset = cmd
+        //     .flags_with_default()
+        //     .into_iter()
+        //     .map(|flag| {
+        //         flag.short.as_ref().map(|s| s.len() + 3).unwrap_or(0)
+        //             + flag.val.as_ref().map(|v| v.name.len() + 3).unwrap_or(0)
+        //             + flag.name.len()
+        //             + 6
+        //     })
+        //     .max()
+        //     .unwrap_or(6);
+        for flag in cmd.flags_with_default() {
             let short = flag.short.as_ref().map(|it| format!("-{it}, ")).unwrap_or_default();
             let value = flag.val.as_ref().map(|it| format!(" <{}>", it.name)).unwrap_or_default();
-            w!(buf, "    {short}--{}{value}\n", flag.name);
-            if let Some(doc) = &flag.doc {
-                write_lines_indented(buf, doc, 6);
-            }
+            let pre_doc = format!("{short}--{}{value}", flag.name);
+            w!(help_buf, "  {:<20} {}\n", pre_doc, flag.doc.as_deref().unwrap_or(""));
         }
     }
-
-    let subcommands = cmd.named_subcommands();
-    if !subcommands.is_empty() {
-        if prefix.is_empty() {
-            blank_line(buf);
-            w!(buf, "SUBCOMMANDS:");
-        }
-
-        let prefix = format!("{}{} ", prefix, cmd.name);
-        for sub in subcommands {
-            blank_line(buf);
-            blank_line(buf);
-            help_rec(buf, &prefix, sub);
-        }
+    w!(help_buf, "\nCommands:");
+    for subcommand in &cmd.subcommands {
+        w!(help_buf, "\n  {:<20} {}", subcommand.name, subcommand.doc.as_deref().unwrap_or(""));
+        let prefix = format!("{}{}__", prefix, subcommand.name);
+        cmd_help_rec(buf, subcommand, &prefix);
     }
+    w!(help_buf, "\n  {:<20} ", "help");
+    w!(help_buf, "Print this message or the help of the given subcommand(s)");
+    w!(buf, "const HELP_{}: &'static str = \"{help_buf}\";\n", snake(prefix).to_uppercase());
 }
 
+// fn w_arg(buf: &mut String, n: usize, msg: &str) {
+//     w!(buf, "  {msg}");
+//     for _ in 0..(n.saturating_sub(msg.chars().count() + 2)) {
+//         buf.push(' ')
+//     }
+// }
```

Comment:
It seems we have a bunch of commented out code

Reply:
Oh I meant to leave that out. I messed around with indenting, didn't even think about wrapping, but committed that anyway. I can maybe revert to 2 commits ago minus outputting the full help message in case a command was supposed to be specified?